From 1dc5633c5b8534a2742823c24238753472e88e3f Mon Sep 17 00:00:00 2001
From: Maxim Ostapenko <maxim.o@samsung.com>
Date: Tue, 28 Jun 2022 19:31:07 +0900
Subject: [PATCH 08/33] Fix deadlock introduced in 05a1f614a8f03e

The join should be done right before exiting main thread.

Signed-off-by: Maxim Ostapenko <maxim.o@samsung.com>
---
 inferenceEngine.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/inferenceEngine.py b/inferenceEngine.py
index af1d8e1..9bb31b4 100644
--- a/inferenceEngine.py
+++ b/inferenceEngine.py
@@ -236,9 +236,9 @@ def inferenceEngine(args,
       if request is None:
         if args.enable_profiling:
             python_profiling_stop()
-        inference_thread.join()
         q_inference_logging.put(None)
         responseQueue.put(None)
+        inference_thread.join()
         return
 
       batch_id   = request.batch_id
-- 
2.34.1

